tests/basic/storage_pool_test.go

```diff
@@ -10626,3 +10626,154 @@ var _ = Describe("{PoolDeleteServiceDisruption}", func() {
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{HAIncreasePoolresizeAndAdddisk}", func() {
+	/*
+		PTX:
+		https://portworx.atlassian.net/browse/PTX-15465
+
+		TestRail:
+		https://portworx.testrail.net/index.php?/cases/view/57783
+
+		 1. Trigger HA increase for volumes
+		 2. Pool resize trigger
+		 3. Add disk to the pool
+		All three steps should occur parallel and should not cause any issue
+	*/
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("HAIncreasePoolresizeAndAdddisk", "HA increase, pool resize and add disk run all this parallely", nil, 57783)
+	})
+	var contexts []*scheduler.Context
+	var nodeToBeUpdated node.Node
+	var poolToBeUpdated string
+	var wg sync.WaitGroup
+
+	itLog := "HAIncreasePoolresizeAndAdddisk"
+	It(itLog, func() {
+		stepLog := "schedule Application"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				for _, app := range Inst().AppList {
+					contexts = append(contexts, ScheduleApplications(fmt.Sprintf("%s-%s", app, "ha-increase-pool-add-resize"))...)
+				}
+			}
+
+		})
+		ValidateApplications(contexts)
+		defer DestroyApps(contexts, nil)
+
+		stepLog = "HA increase for volumes"
+		Step(stepLog, func() {
+			// First get volumes of the application
+			for _, eachContext := range contexts {
+				vols, err := Inst().S.GetVolumes(eachContext)
+				log.FailOnError(err, "Failed to get volumes from context")
+				//pick a random volume
+				vol := vols[rand.Intn(len(vols))]
+				curReplSet, err := Inst().V.GetReplicationFactor(vol)
+				log.FailOnError(err, "failed to get replication factor of the volume")
+				// Check if Replication factor is 3. if so, then reduce the repl factor and then set repl factor to 2
+				if curReplSet == 3 {
+					newRepl := int64(curReplSet - 1)
+					log.FailOnError(Inst().V.SetReplicationFactor(vol, newRepl,
+						nil, nil, true),
+						"Failed to set Replicaiton factor")
+				}
+				// now increase the replication factor of this volumes.
+				nodeToBeUpdated = node.GetStorageDriverNodes()[rand.Intn(len(node.GetStorageDriverNodes()))]
+				pools, err := GetAllPoolsOnNode(nodeToBeUpdated.Id)
+				poolToBeUpdated = pools[0]
+				log.FailOnError(err, "Failed to get all pools on node: %v", nodeToBeUpdated.Name)
```

Comment:
Move this line right after GetAllPoolsOnNode

Reply:
Done